math/src/test/java/eu/itesla_project/math/matrix/PlainMatrixTest.java

```diff
@@ -70,4 +65,12 @@ public void testCreateFromColumn() {
         assertEquals(2d, a.getValue(1, 0), 0d);
         assertEquals(3d, a.getValue(2, 0), 0d);
     }
+
+    @Test
+    public void testToSparse() {
+        PlainMatrix a = (PlainMatrix) createA(matrixFactory);
+        SparseMatrix a2 = a.toSparse();
+        PlainMatrix a3 = a2.toPlain();
+        assertEquals(a, a3);
```

Comment:
equals is implemented just for testing as it would not be reasonable to perform comparison on real sized matrix. Choice here is to implement type by type comparison.